apps/src/netsim/NetSimRouterNode.js

```diff
@@ -657,3 +686,69 @@ NetSimRouterNode.prototype.routeMessage_ = function (message, myWires) {
       }.bind(this)
   );
 };
+
+/**
+ * @param {NetSimMessage} message
+ * @param {NetSimWire[]} myWires
+ * @private
+ */
+NetSimRouterNode.prototype.generateDnsResponse_ = function (message, myWires) {
+  var fromAddress, query;
+
+  // Extract message contents
+  try {
+    fromAddress = dataConverters.binaryToInt(
+        PacketEncoder.defaultPacketEncoder.getField('fromAddress', message.payload));
+    query = dataConverters.binaryToAscii(
+        PacketEncoder.defaultPacketEncoder.getField('message', message.payload),
+        BITS_PER_BYTE);
+  } catch (error) {
+    // Malformed packet, ignore
+    return;
+  }
+
+  // Check that the query is well-formed
+  var requestMatch = query.match(/GET\s+(\S.*)/);
```

Comment:
Well *I* do of course, but I'm cognizant of the fact that not everyone else
does :-P

On Thu, Mar 12, 2015 at 12:46 PM, Brad Buchanan <notifications@github.com>
wrote:

> In apps/src/netsim/NetSimRouterNode.js
> <https://github.com/code-dot-org/code-dot-org/pull/1585#discussion_r26337462>
> :
>
> > +  var fromAddress, query;
> > +
> > +  // Extract message contents
> > +  try {
> > +    fromAddress = dataConverters.binaryToInt(
> > +        PacketEncoder.defaultPacketEncoder.getField('fromAddress', message.payload));
> > +    query = dataConverters.binaryToAscii(
> > +        PacketEncoder.defaultPacketEncoder.getField('message', message.payload),
> > +        BITS_PER_BYTE);
> > +  } catch (error) {
> > +    // Malformed packet, ignore
> > +    return;
> > +  }
> > +
> > +  // Check that the query is well-formed
> > +  var requestMatch = query.match(/GET\s+(\S.*)/);
>
> What? You don't live and breathe /[regula]{7}\s+[exprsion]{11}/i?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/1585/files#r26337462>.
>